packages/optimizer/lib/transformers/MinifyHtml.js

```diff
@@ -77,8 +75,11 @@ class MinifyHtml {
     if (opts.canCollapseWhitespace && !this.canCollapseWhitespace(node.tagName)) {
       childOpts.canCollapseWhitespace = false;
     }
-    if (opts.canTrimWhitespace && !this.canTrimWhitespace(node.tagName)) {
-      childOpts.canTrimWhitespace = false;
+    if (node.tagName === 'head' || node.tagName === 'html') {
+      childOpts.inBody = false;
+    }
+    if (node.tagName === 'body') {
```

Comment:
The entire `isBody` logic to decide whether to trim would fail if there is no `<body>` tag in the document. Is it assumed the `<body>` tag exists because of previous transformers?

Reply:
Yes, the AddMandatoryTag transformer will add the body.  Otherwise, I'd prefer to err on the save side and not trim whitespaces.